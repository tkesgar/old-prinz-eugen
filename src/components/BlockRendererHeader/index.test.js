import { mount } from '@vue/test-utils'
import localVue from '../../utils/test/local-vue'
import BlockRendererHeader from './'

describe('BlockRendererList', () => {
  const text = 'Prinz Eugen'
  const wrapper = mount(BlockRendererHeader, {
    localVue,
    propsData: {
      data: { text, level: 1 }
    }
  })

  test('should match snapshot', () => {
    expect(wrapper.element).toMatchSnapshot()
  })

  test('should render <h1>', () => {
    expect(wrapper.find('h1').exists()).toBe(true)
  })

  test('should render text', () => {
    expect(wrapper.find('h1').text()).toBe(text)
  })
})

describe('BlockRendererList test with all heading levels', () => {
  for (let level = 1; level <= 6; level++) {
    test(`should render h${level}`, () => {
      const wrapper = mount(BlockRendererHeader, {
        localVue,
        propsData: {
          data: { text: 'Prinz Eugen', level }
        }
      })

      expect(wrapper.find(`h${level}`).exists()).toBe(true)
    })
  }
})
