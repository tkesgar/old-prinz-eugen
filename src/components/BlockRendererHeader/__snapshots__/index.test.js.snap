// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlockRendererList should match snapshot 1`] = `
<h1>
  Prinz Eugen
</h1>
`;
